test-suite/misc/changelog.sh

```diff
@@ -9,7 +9,8 @@ done < ../config/coq_config.py
 
 for d in ../doc/changelog/*; do
   if [ -d "$d" ]; then
-    if [ "$(ls $d/*.rst | wc -l)" != "1" ]; then
+    if [ "$(ls -1 $d/*.rst | wc -l)" != "1" ]; then
+      echo "$(ls -1 $d/*.rst | wc -l)"
```

Comment:
Using ls for this seems wrong
How about 
~~~bash
        files=("$d"/*.rst)
        if [ "${#files[@]}" != 1 ]; then
            echo "${files[@]}"
~~~
? (you'll need to change the shebang to use bash)

or 
~~~sh
        set - "$d"/*.rst
        if [ "$#" != 1 ]; then
            echo "$@"
~~~
if you want to keep using posix shell

Reply:
Thanks!